app/main.py

```diff
@@ -1,2 +1,50 @@
+from typing import Any
+
+
 class Dictionary:
-    pass
+    def __init__(self, capacity: int = 8) -> None:
+        self.capacity = capacity
+        self.size = 0
+        self.table = [[] for _ in range(self.capacity)]
+
+    def get_hash(self, key: Any) -> int:
+        return hash(key) % self.capacity
+
+    def __setitem__(self, key: Any, value: Any) -> None:
+        hash_index = self.get_hash(key)
+        slot = self.table[hash_index]
+
+        for index, (existing_key, existing_value) in enumerate(slot):
+            if existing_key == key:
+                slot[index] = (key, value)
+                break
+        else:
+            slot.append((key, value))
+            self.size += 1
+
+        if self.size >= self.capacity * 2:
+            self.resize()
+
+    def __getitem__(self, key: Any) -> Any:
+        hash_index = self.get_hash(key)
+        slot = self.table[hash_index]
+
+        for existing_key, existing_value in slot:
+            if existing_key == key:
+                return existing_value
+
+        raise KeyError(f"Key '{key}' not found")
```

Comment:
You can enhance the efficiency of your code by implementing an approach that directly calculates the index based on the key and retrieves the value from the hash table. This approach avoids unnecessary iterations and checks, making it a more efficient way to access key-value pairs

Reply:
you didn't fix it. Find values associated with keys by directly using the key's hash code for quick identification and retrieval, without the need for time-consuming for loops